src/net.c

```diff
@@ -374,6 +375,32 @@ int valkeyTcpSetTimeout(valkeyContext *c, const struct timeval tv) {
     return VALKEY_OK;
 }
 
+#ifdef IPPROTO_MPTCP
+int valkeyHasMptcp(void) {
+    return 1;
+}
+
+/* XXX: Until glibc 2.41, getaddrinfo with hints.ai_protocol of IPPROTO_MPTCP leads error.
+ * Use hints.ai_protocol IPPROTO_IP (0) or IPPROTO_TCP (6) to resolve address and overwrite
+ * it when MPTCP is enabled.
+ * Ref: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/testing/selftests/net/mptcp/mptcp_connect.c
+ *      https://sourceware.org/git/?p=glibc.git;a=commit;h=a8e9022e0f829d44a818c642fc85b3bfbd26a514
+ */
+static int valkeyTcpGetProtocol(int is_mptcp_enabled) {
+    return is_mptcp_enabled ? IPPROTO_MPTCP : IPPROTO_TCP;
+}
```

Comment:
I think it would be easier to read if we ifdef within the functions, WDYT?
```suggestion
static int valkeyTcpGetProtocol(int is_mptcp_enabled) {
#ifdef IPPROTO_MPTCP
    return is_mptcp_enabled ? IPPROTO_MPTCP : IPPROTO_TCP;
#else
    assert(!is_mptcp_enabled);
    return IPPROTO_TCP;
#endif /* IPPROTO_MPTCP */
}
```

Same comment for `valkeyHasMptcp`.

Reply:
This is the previous version, I converted to current version. Hahaha~
I'm not sure which style is better...

Reply:
Someday if `IPPROTO_MPTCP` is popular enough, just removing the `#else` branch is OK...

Reply:
I think it doesn't matter. Both are OK. I would leave it up to the author. :)

But we need to keep `#else` for a long time, because we support many platforms and some of them will probably never have MPTCP within a near future.

Reply:
This was just a minor nit to have a similar look as its (currently) done in valkey. Current is fine.